extension.js

```diff
@@ -14,9 +14,42 @@ const EventEmitter = require('events');
 let preferences = require("./preferencesHelper");
 const writeToTestCaseFile = require("./writeToTestCaseFile");
 const handleCompanion = require("./companionHandler");
+const config = require('./config');
+
 let oc = vscode.window.createOutputChannel("competitive");
 let spawnStack = [];
 
+/**
+ * Retrieve extension of file
+ * @param {string} filePath complete path to .c, .cpp or .py file
+ */
+function getExtension(filePath) {
+  const fileExtension = filePath
+    .split(".")
+    .pop()
+    .toLowerCase();
+  return fileExtension;
+}
+
+/**
+ * Get language based on file extension
+ * @param {string} extension c, cpp or py
+ */
+function getLanguage(extension) {
+  for (const [lang, ext] of Object.entries(config.extensions))
```

Comment:
Let's throw an error here if no matching extension is found. Although it will likely never happen, because we are aleady checking it in line 428.

Reply:
I'm not agree with throw an error, I think it's better to returns `undefined` which means not found and each module which use it is responsible for it (e.g. show a message about it)

Reply:
I refactored line 428. Let's check it!

Reply:
Okay.